packages/solid/src/reactive/signal.ts

```diff
@@ -592,10 +654,10 @@ export function createSelector<T, U>(
  *
  * @description https://www.solidjs.com/docs/latest/api#batch
  */
-export function batch<T>(fn: () => T): T {
+export function batch<T>(fn: Accessor<T>): T {
```

Comment:
Not sure if `Accessor` makes sense in this case, since I don't think you'd ever pass an accessor to batch. It might make more sense to leave this as `() => T`.

Reply:
Does it expect any function, not necessarily a reactive getter?

Reply:
I think usually it would be a void function containing a few set calls, possibly returning a value if convenient to do so. Either way, passing a reactive getter i.e. `batch(getter)` doesn't make sense compared to something like `untrack(getter)`.